crates/ibc/src/core/context.rs

```diff
@@ -90,14 +90,14 @@ pub trait ValidationContext {
     ) -> Result<Option<Box<dyn ConsensusState>>, ClientError>;
 
     /// Returns the current height of the local chain.
-    fn host_height(&self) -> Height;
+    fn host_height(&self) -> Result<Height, ClientError>;
 
     /// Returns the current timestamp of the local chain.
-    fn host_timestamp(&self) -> Timestamp {
+    fn host_timestamp(&self) -> Result<Timestamp, ClientError> {
```

Comment:
`ClientError` is not appropriate here, but we don't have a good solution for that at the moment. We would really need some `HostError`. We can fix this with #164.

Reply:
I would also like to use this way to solve, it is not very clear that the current no_std support is not very perfect